### DISCUSSION TITLE
7.1.4

### DISCUSSION THREAD

**Discussion by Ghost-chu** [PBH-BTN/PeerBanHelper#724] [公告]
## 反吸血算法优化

* 优化进度反作弊检测模块对使用 “超级做种” 的 Peer 的误封禁的问题。现在在对方未从 PBH 所关联的下载器下载数据的情况下，PCB 进度反作弊检测模块将不会动作，以避免错误的封禁使用 [BEP-0016](https://www.bittorrent.org/beps/bep_0016.html) 规范进行超级做种的 Peer（超级做种启用时，将不会汇报真实的任务进度给其它人） @Ghost-chu 

## 默认配置文件调整

自 2024 年年初以来，已经度过了快一年了。在军备竞赛中火拼了数月后，吸血 Peer 的特征经历了一波又一波的变化。PBH 自带的部分配置已不再适应目前的现状，我们对不再需要的规则进行了删除，以下是删除的清单：

- 删除 `anacrolix/torrent` 和 `-GT0003` 规则，现在的吸血客户端已不再使用 `anacrolix/torrent` 进行吸血，转为伪装其它正规客户端。根据 BTN 收集的数据来看，此规则挡住了一些基于 anacrolix/torrent 开发的正规客户端（如番剧下载器等），因此我们删除了此规则
  - 后续如果卷土重来，将通过 BTN 云端规则下发重新屏蔽
- 删除 `go.torrent` 和 `-GT0002` 规则，在目前 BT 环境变得愈加恶劣的情况下，百度网盘的离线下载功能能够起到有效分流、减少流量上传的作用，且经常被用于救种。值得一提的是，我们注意到百度网盘的离线下载在下载期间会上传数据，这有助于 BT 文件分发并改善 Peer 之间的连接性；综上所述，我们没有理由再继续屏蔽百度网盘的离线下载功能
- 删除 `ljyun.cn/hangzhou/monitoring` 规则，因为这家边缘云厂商现在显然不再使用这个 ClientName 下载文件了

已经安装过 PBH 的用户需要手动进行上述修改，该更改不会自动应用到现有安装。

## 错误修复

* 修复 qBittorrent EE 连接到 Web Seed 时报告 IP 地址转换错误的问题 @Ghost-chu 
* 修复 BTN 模块中存在的一个问题导致 PBH 不会执行 BTN 规则的问题 @Ghost-chu 
* 修复 BTN 云端规则未获取完成的情况下，状态面板的 BTN 状态可能显示为空 @Ghost-chu 

## Docker

DockerHub: `ghostchu/peerbanhelper:v7.1.4`
阿里云国内镜像加速: `registry.cn-hangzhou.aliyuncs.com/ghostchu/peerbanhelper:v7.1.4`

<hr /><em>This discussion was created from the release <a href='https://github.com/PBH-BTN/PeerBanHelper/releases/tag/v7.1.4'>7.1.4</a>.</em>

**Comment by AmeroHan** [comment]
我想询问一下，百度网盘和迅雷的离线下载区别是否只有下载期间会上传数据？如果百度网盘在下载期间没有其他peer需要下载，那百度和迅雷的区别呢？如果需要救种，说明资源很冷门，那么没有其他peer需要下载的可能性是极大的；在资源热门时，离线下载的确可以起到分流的作用，但即使PBH不解除屏蔽，一样有大量未安装PBH的peer来为百度网盘上传。

**Reply by Nahuimi** [reply]
我的理解是迅雷作为下载器，他的用户是可以从其他peer吸血的，但是百度云他的用户不行，百度云它离线成功一次，以后它的用户离线下载都是秒成功，取回文件都是用户从百度云自己的服务器下载。

**Reply by xhxcxrthebtyuhjyu8h** [reply]
如果百度云真的是全部依靠自己的带宽的话，可能作为一个种子备份库是挺不错的。不过鉴于这个网盘那超严格的审核，可能对p2p分享环境没什么帮助吧

**Comment by Nahuimi** [comment]
其他的网盘离线的规则是怎么样的，例如115，夸克，PikPak之类的也都是放行吗？

**Reply by Ghost-chu** [reply]
保持现状

**Comment by sink-er** [comment]
对于已经安装过 PBH 的用户，要怎么手动修改配置文件

**Reply by Duck1998** [reply]
管理端-设置-首选项-模块配置-PeerId封禁和客户端名称封禁，自己删，最后记得拉到网页底部点保存
